spec/admin_spec.rb

```diff
@@ -0,0 +1,88 @@
+require_relative 'spec_helper'
+
+describe "#initialize" do
+  it "Initializes a class variable that stores an empty string" do
+    test = Admin.new
+
+    expect(test).must_respond_to :blocked_start_date
+    expect(test.blocked_start_date).must_be_nil
+
+    expect(test).must_respond_to :blocked_end_date
+    expect(test.blocked_start_date).must_be_nil
+  end
+ end
+
+describe "block rooms" do
+
+  it "Returns an array of blocked rooms" do
+    test = Admin.new
+    check = test.block_rooms(3, "2018-10-01", "2018-10-15")
+
+      expect(check.length).must_equal 3
+  end
+end
+
+describe "Checks if blocked room is booked" do
+  it "Confirms a room from blocked rooms is booked" do
+    test = Admin.new
+    check = test.block_rooms(3, "2018-10-01", "2018-10-15")
+    new = test.book_blocked_room
+
+    expect(new.length).must_equal 2
+  end
+end
+
+describe "Checks rooms array " do
+  it "Checks length of array containing rooms" do
+    test = Admin.new
+    list = test.view_all_rooms
+
+    expect(list.length).must_equal 20
+  end
+end
+
+describe "Prints cost per reservation" do
+  it "Checks the total cost for a reservation" do
+    person = Admin.new
+    person.reserve(3, "2018-11-01", "2018-11-15")
+    pay = person.print_total_cost_per_reservation
+
+    expect(pay).must_equal 2600
```

Comment:
This expect is based on incorrect data. If we are counting each night, then there are 14 nights to be accounted for between the 1st and the 15th. If each costs 200, then the expected cost would be 2800. When I look at this, I get the feeling that the test was written after the code, because the answer that you expect matches the error made inside your class.